backpack-common/src/main/java/net/skyscanner/backpack/calendar2/data/CalendarStateMachine.kt

```diff
@@ -121,14 +121,14 @@ internal fun CalendarState.dispatchSetSelection(selection: CalendarSelection): C
     when (selection) {
         is CalendarSelection.None -> Unit
         is CalendarSelection.Dates -> when {
-            params.selectionMode != SelectionMode.Range -> return this
+            params.selectionMode != SelectionMode.Range() -> return this
```

Comment:
You are basically comparing it here with an empty data class, which will break calendar logic if accessible strings are provided. The correct syntax is `params.selectionMode !is SelectionMode.Range`

Reply:
Fixed, thanks!